index.bs

```diff
@@ -781,23 +843,41 @@ with parameter |value| is:
 
 <div algorithm="webSocketUrl new session algorithm">
 The [=WebDriver new session algorithm=] defined by this specification,
-with parameters |session| and |capabilities| is:
+with parameters |session|, |capabilities|, and |flags| is:
+
+1. If |flags| contains "<code>bidi</code>", return.
+
+1. Let |webSocketUrl| be the result of [=getting a property=] named
+   "<code>webSocketUrl</code>" from |capabilities|.
 
- 1. Let |webSocketUrl| be the result of [=getting a property=] named
-    "<code>webSocketUrl</code>" from |capabilities|.
+1. If |webSocketUrl| is undefined, or false, return.
 
- 2. If |webSocketUrl| is undefined, return.
+1. [=Assert=]: |webSocketUrl| is true.
```

Comment:
Shouldn't we specify the type of error to be returned if it fails?

Reply:
There is no error. If the flag isn't set we didn't request a BiDi session, so we don't do anything for BiDi but could still create a HTTP-only session.

Reply:
Right, but that is covered by the former check on line 874. Here it would still be possible to pass in any type, which might not be a boolean. As such we can fail.

Reply:
I'm not sure if `webSocketUrl` is guaranteed to be true, but if it's not we shouldn't use an assert here at all.

Reply:
It is, hence the assert :) If `webSocketUrl` was set to something other than a boolean we'd fail during capabilites validation, see the `webSocketUrl capability deserialization algorithm` slightly further up.